File: Scripts/laptop/factorial/c/factorial.c
#include <stdio.h>

int factorial(int n);

int main() {

volatile int result;

for (int i = 0; i < 2000000000; ++i) {
	result = executeTask(i + result);
}

return 0;
}

int executeTask(int i){
	factorial(10);
	return i;
}

int factorial(int n) {
    return n == 0 ? 1 : n * factorial(n - 1);
}

File: Scripts/laptop/factorial/javascript/factorial.js
var num = 10;

var factorial = function(n) {
    if(n == 0) {
        return 1
    } else {
        return n * factorial(n - 1);
    }
}

var r = 1;

function executeTask(i) {

	factorial(num);
	return i + 1;
}
 
for (var i = 0; i < 2000000000; ++i) { 
	r = executeTask(i + r);
}
